### DISCUSSION TITLE
Side-by-side ggplots with working cross-reference and title

### DISCUSSION THREAD

**Discussion by RohanAlexander** [quarto-dev/quarto-cli#684] [Q&A] [answered]
This builds on #590 

In the accepted answer both graphs have been given the same caption (the text from 'fig-cap'). And adding a cross-reference doesn't seem to work (see image below code).

`
@fig-TESTING

```{r}
#| label: fig-TESTING
#| fig-cap: "A test plot."
#| layout: "[1, 1]"

library(ggplot2)
plt <- ggplot(mtcars, aes(mpg, wt, colour = factor(cyl))) + 
  geom_point(size = 3) + 
  theme_classic()

plt
plt2 <- ggplot(airquality, aes(Wind, Temp)) + 
  geom_point() + 
  theme_classic()

plt2
```
`

<img width="1080" alt="Screen Shot 2022-04-19 at 9 11 36 AM" src="https://user-images.githubusercontent.com/5662323/164014458-b078b2f6-b5ef-43c5-a272-e299f5b4665a.png">

Does anyone has qmd code that adds just one overall caption to the figure, rather than duplicating the same caption for each figure (which I'm hoping would fix the cross-reference issue) please? Or alternatively, allows for a separate caption and label for each side-by-side figure?

**Comment by dragonstyle** [comment, marked as answer]
One approach would be to use sub figures. This would allow you to provide captions for the overall figure as well as the individual plots. In addition, your cross reference can then refer to the whole figure or either subfigure. For example:

````markdown
---
title: A Simple Plot Example
format: html
---

```{r}
#| label: fig-testing
#| fig-cap: "A pair of plots, laid out side by side."
#| fig-subcap: ["Cars","Air Quality"]
#| layout: "[1, 1]"
library(ggplot2)
plt <- ggplot(mtcars, aes(mpg, wt, colour = factor(cyl))) + 
  geom_point(size = 3) + 
  theme_classic()

plt
plt2 <- ggplot(airquality, aes(Wind, Temp)) + 
  geom_point() + 
  theme_classic()

plt2
```

The overall figure @fig-testing


The first sub figure @fig-testing-1


The second subfigure @fig-testing-2

````

which renders as:

![Screen Shot 2022-04-19 at 2 15 27 PM](https://user-images.githubusercontent.com/261654/164069338-498652e2-4a6b-45ce-8b01-599e74f4b721.png)

Would this work for you?

**Reply by RohanAlexander** [reply]
This is perfect. I'm sorry I couldn't work this out myself. Thank you very much for your quick answer.

**Reply by dragonstyle** [reply]
No worries at all! I'm glad it worked well for you - Quarto has a lot of capabilities so it can be hard to know what you're looking for - we will keep improving the education and documentation around it - questions and answers like this help too, so thank you!
